### PR TITLE
make version required argument to avoid confusion and make readme exa…

### PR DESCRIPTION
**Pull request by grosser**
…mples consistent

there is more work in this area, like `'host/apis/batch', 'v1'` being weird
but I think this is a nice start on removing confusion

### CODE REVIEW

**Comment by grosser** [comment]
@cben thoughts on this ?

**Comment by cben** [comment]
Not a bad idea.
This doesn't gain much, but it small positive change towards untangling the API groups mess :+1:.

- One question I like to ask about breaking changes is whether the required rewrite for users is tightly coupled to the upgrade.  It's better when there is a major version which supports both old and new interface — this allows piece-wise rewrites.
  This satisfies that because people can add explicit `, 'v1'` in previous version :heavy_check_mark:. 

I'd like to hold this back for now and decide close to 5.0 :thinking: release but I'm inclined to include it.

- [ ] We'll need to list this in CHANGELOG, and in "Upgrading past version 5.0" section of README.
- [ ] What happens if someone runs old code omitting version?  Is the error message clear?

**Comment by cben** [comment]
(and yes, I'm totally over-thinking this release :laughing:)

**Comment by grosser** [comment]
- prefer no Upgrading section in the readme, that's for the changelog and gets out of date quick leaving obsolete info in the readme
- users that omit it get a nice error message (newly added raise)

**Comment by cben** [comment]
OK, let's remove the existing Upgrading section then.

**Comment by grosser** [comment]
https://github.com/abonas/kubeclient/pull/476

**Comment by grosser** [comment]
added changelog

File: test/test_kubeclient.rb [hunk 1]
```diff
@@ -34,18 +34,12 @@ def test_no_path_in_uri
     assert_equal('http://localhost:8080/api/v1', rest_client.url.to_s)
   end
 
-  def test_no_version_passed
```
Comment (astencel-sumo) [inline test/test_kubeclient.rb:37]: How about we leave this test in and assert it raises that ArgumentError?

Comment (grosser) [reply test/test_kubeclient.rb:39]: done

Comment (astencel-sumo) [reply test/test_kubeclient.rb:39]: Rubocop complaining now 👮

Comment (grosser) [reply test/test_kubeclient.rb:39]: fixed!

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,10 @@ Notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 Kubeclient release versioning follows [SemVer](https://semver.org/).
 
+## Unreleased
+### Changed
+- `Kubeclient::Client.new` always required an api version, use for example: `Kubeclient::Client.new(uri, 'v1')`
```
Comment (cben) [inline CHANGELOG.md:9]:
```suggestion
- `Kubeclient::Client.new` now always requires an api version, use for example: `Kubeclient::Client.new(uri, 'v1')`
```

**Comment by cben** [comment]
Yeah, let's do this :rocket:.  Please resolve conflict.

**Comment by grosser** [comment]
rebased

**Pull request merged**